multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/MergeDescriptorsStep.java

```diff
@@ -32,10 +41,39 @@ protected StepPhase executeStep(ProcessContext context) {
         DeploymentDescriptor descriptor = getMtaDescriptorMerger(handlerFactory, platform).merge(deploymentDescriptor,
                                                                                                  extensionDescriptors);
         context.setVariable(Variables.DEPLOYMENT_DESCRIPTOR, descriptor);
+        context.setVariable(Variables.CHECKSUM_OF_MERGED_DESCRIPTOR, DigestUtils.md5DigestAsHex(JsonUtil.toJson(descriptor)
+                                                                                                        .getBytes()));
+        preserveDeploymentDescriptor(context, descriptor);
         getStepLogger().debug(Messages.DESCRIPTORS_MERGED);
         return StepPhase.DONE;
     }
 
+    private void preserveDeploymentDescriptor(ProcessContext context, DeploymentDescriptor descriptor) {
+        if (!context.getVariable(Variables.SHOULD_PRESERVE_OLD_APPS)) {
+            return;
+        }
```

Comment:
can this be extracted in the diagram?

Reply:
How? This step is used in all different type of deployment operations.